### ISSUE TITLE
Sortable Date Columns

### ISSUE THREAD

**Issue by Bush-cat** [w-ahmad/WinUI.TableView#26]
**Is your feature request related to a problem? Please describe.**
I use a localized DateOnly in my Table but Sorting is broken because of the Date format DD/MM/YYYY.

**Describe the solution you'd like**
Make TextColumns respect Date classes or maybe make a  DateColumn that can work with Date variables

**Describe alternatives you've considered**
Some way to implement sorting into a TextColumn yourself, this would also be usefull in other Cases, where I have an Icon next to my Text, there I couldn't use a TextColumn and decided to use two Columns instead, one for the icon, the other for the text.

**Comment by VisualAlf** [comment]
IMHO the easiest way for this is to use a converter.
If you have eg.  an integer as the underlying datatype it will be sorted correctly, regardless of the displayed text format.

**Comment by Bush-cat** [comment]
> IMHO the easiest way for this is to use a converter. If you have eg. a numeric as the underlying datatype it will be sorted correctly, regardless of the displayed text format. See the SampleApp for usage with a DateTimeOffset.

Converter could work, sadly Converters do not work with the Filter Menu, it shows the items without any conversion
Maybe the new template controls in v1.2.0-preview2 can do it. More TemplateColumn properties would help a lot.
I'm a bit struggling with localization in winui, I tried many ways already but they all suck in some way.

**Comment by w-ahmad** [comment]
You are correct, @Bush-cat; converters are not considered for sorting and filtering. Would it be suitable for you if the filter and sort methods use converters to get the final values?

**Comment by Bush-cat** [comment]
> You are correct, @Bush-cat; converters are not considered for sorting and filtering. Would it be suitable for you if the filter and sort methods use converters to get the final values?

Converters for those would be nice
